### PR TITLE
[Inductor Intel GPU backend Upstream] Generalize part of Inductor test case

### PR DESCRIPTION
**Pull request by etaf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117513

Following the RFC https://github.com/pytorch/pytorch/issues/114856, before upstream Intel XPU Inductor Backend, we need to preapre corresponding Inductor test cases. This PR aims to generalize part of Inductor test case so that a new GPU backend can reuse the existing test case with minimal code change.

This Pull Request preferentially generalizes the test cases that cover Inductor's base functionality as follow:
- test/inductor/test_codecache.py
- test/inductor/test_codegen_triton.py
- test/inductor/test_kernel_benchmark.py
- test/inductor/test_torchinductor.py
- test/inductor/test_torchinductor_codegen_dynamic_shapes.py
- test/inductor/test_torchinductor_dynamic_shapes.py
- test/inductor/test_torchinductor_opinfo.py
- test/inductor/test_triton_heuristics.py
- test/inductor/test_triton_wrapper.py

Feature request: https://github.com/pytorch/pytorch/issues/114856

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by EikanWang — CHANGES_REQUESTED** [review]

**Comment by EikanWang** [comment]
Overall, LGTM.

File: test/inductor/test_codecache.py [hunk 1]
```diff
@@ -227,14 +227,14 @@ def fn(x, y):
...
-    @parametrize("device", ("cuda", "cpu"))
+    @parametrize("device", (GPU_TYPE, "cpu"))
     @parametrize("dtype", (torch.float32, torch.bfloat16))
     def test_cache_load_with_guards_static_bounds(self, device, dtype):
         """
         Test caching the same graph, but under conditions that introduce guards
         for static bounds.
         """
-        if device == "cuda" and not HAS_CUDA:
+        if device in GPU_TYPE and not HAS_GPU:
```
[the remark this answers is not in the document]
Comment (etaf) [reply test/inductor/test_codecache.py:237]: Yes. Sorry for the typo.

File: test/inductor/test_torchinductor.py [hunk 2]
```diff
@@ -93,21 +92,26 @@
...
 
 HAS_AVX2 = "fbgemm" in torch.backends.quantized.supported_engines
 _desired_test_bases = get_desired_device_type_test_bases()
 RUN_CPU = any(getattr(x, "device_type", "") == "cpu" for x in _desired_test_bases)
-RUN_CUDA = any(getattr(x, "device_type", "") == "cuda" for x in _desired_test_bases)
+RUN_GPU = any(getattr(x, "device_type", "") == GPU_TYPE for x in _desired_test_bases)
 
 aten = torch.ops.aten
 requires_cuda = functools.partial(unittest.skipIf, not HAS_CUDA, "requires cuda")
+requires_gpu = functools.partial(unittest.skipIf, not HAS_GPU, "requires gpu")
```
[the remark this answers is not in the document]
Comment (etaf) [reply test/inductor/test_torchinductor.py:111]: Yes, I will remove this. The reason I keep it here is that  other two test file that we did not cover yet, will import `test_torchinductor.requires_cuda`. I would like to update the other two file.

**Review by EikanWang — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Comment by jeanschmidt** [comment]
I am investigating the breakage of `pull / linux-focal-cuda12.1-py3.10-gcc9-sm86 / test` on main. I suspect that the origin is not the PR itself, but instead how tests are grouped in chunks. Reaching out to the respective team.

**Pull request closed without merging**